### ISSUE TITLE
[Bug]: some collections load(load after release) timeout after milvus recover from many pod kill/failure chaos test

### ISSUE THREAD

**Issue by zhuwenxing** [milvus-io/milvus#31162]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:master-20240308-c8efed65
- Deployment mode(standalone or cluster):cluster
- MQ type(rocksmq, pulsar or kafka):    
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

```
[2024-03-10T20:24:54.895Z] [2024-03-10 20:21:30 - INFO - ci_test]: index info: [{'collection': 'QueryChecker__gxnVwxgP', 'field': 'text_emb', 'index_name': 'text_emb', 'index_param': {'index_type': 'HNSW', 'metric_type': 'L2', 'params': {'M': 48, 'efConstruction': 500}}}, {'collection': 'QueryChecker__gxnVwxgP', 'field': 'voice_emb', 'index_name': 'voice_emb', 'index_param': {'index_type': 'HNSW', 'metric_type': 'L2', 'params': {'M': 48, 'efConstruction': 500}}}, {'collection': 'QueryChecker__gxnVwxgP', 'field': 'int64', 'index_name': 'int64', 'index_param': {'index_type': 'INVERTED'}}, {'collection': 'QueryChecker__gxnVwxgP', 'field': 'float', 'index_name': 'float', 'index_param': {'index_type': 'INVERTED'}}, {'collection': 'QueryChecker__gxnVwxgP', 'field': 'varchar', 'index_name': 'varchar', 'index_param': {'index_type': 'INVERTED'}}, {'collection': 'QueryChecker__gxnVwxgP', 'field': 'float_vector', 'index_name': 'float_vector', 'index_param': {'index_type': 'HNSW', 'metric_type': 'L2', 'params': {'M': 48, 'efConstruction': 500}}}, {'collection': 'QueryChecker__gxnVwxgP', 'field': 'image_emb', 'index_name': 'image_emb', 'index_param': {'index_type': 'HNSW', 'metric_type': 'L2', 'params': {'M': 48, 'efConstruction': 500}}}] (test_all_collections_after_chaos.py:86)

[2024-03-10T20:24:54.895Z] [2024-03-10 20:21:30 - DEBUG - ci_test]: (api_request)  : [Collection.load] args: [None, 1, 180], kwargs: {} (api_request.py:62)

[2024-03-10T20:24:54.895Z] [2024-03-10 20:21:30 - DEBUG - ci_test]: (api_response) : None  (api_request.py:37)

[2024-03-10T20:24:54.895Z] [2024-03-10 20:21:30 - DEBUG - ci_test]: (api_request)  : [Collection.search] args: [[[0.1372567886185564, 0.1364428530633049, 5.475124225995655e-05, 0.022193052852356576, 0.13329689134391545, 0.1256094460303883, 0.019190952927515107, 0.11397607230318842, 0.09947889011072082, 0.13223848736105875, 0.12021330344448018, 0.034058508829838, 0.1159998653073527, 0.05041840383575526, 0.128......, kwargs: {} (api_request.py:62)

[2024-03-10T20:24:54.895Z] [2024-03-10 20:21:30 - DEBUG - ci_test]: (api_response) : ["['id: 7287024119668426513, distance: 23.48675537109375, entity: {}']"]  (api_request.py:37)

[2024-03-10T20:24:54.895Z] [2024-03-10 20:21:30 - INFO - ci_test]: assert search: 0.27668166160583496 (test_all_collections_after_chaos.py:99)

[2024-03-10T20:24:54.895Z] [2024-03-10 20:21:30 - DEBUG - ci_test]: (api_request)  : [Collection.query] args: ['int64 in [-3000, -2999, -2998, -2997, -2996, -2995, -2994, -2993, -2992, -2991, -2990, -2989, -2988, -2987, -2986, -2985, -2984, -2983, -2982, -2981, -2980, -2979, -2978, -2977, -2976, -2975, -2974, -2973, -2972, -2971, -2970, -2969, -2968, -2967, -2966, -2965, -2964, -2963, -2962, -2961, -2960, -......, kwargs: {} (api_request.py:62)

[2024-03-10T20:24:54.895Z] [2024-03-10 20:21:31 - DEBUG - ci_test]: (api_response) : [{'int64': -3000}, {'int64': -2999}, {'int64': -2998}, {'int64': -2997}, {'int64': -2996}, {'int64': -2995}, {'int64': -2994}, {'int64': -2993}, {'int64': -2992}, {'int64': -2991}, {'int64': -2990}, {'int64': -2989}, {'int64': -2988}, {'int64': -2987}, {'int64': -2986}, {'int64': -2985}, {'int64': -......  (api_request.py:37)

[2024-03-10T20:24:54.895Z] [2024-03-10 20:21:31 - INFO - ci_test]: assert query result 2000: 0.4092884063720703 (test_all_collections_after_chaos.py:106)

[2024-03-10T20:24:54.895Z] [2024-03-10 20:21:31 - DEBUG - ci_test]: (api_request)  : [Collection.release] args: [], kwargs: {'timeout': 180} (api_request.py:62)

[2024-03-10T20:24:54.895Z] [2024-03-10 20:21:32 - DEBUG - ci_test]: (api_response) : None  (api_request.py:37)

[2024-03-10T20:24:54.895Z] [2024-03-10 20:21:35 - DEBUG - ci_test]: (api_request)  : [Collection.insert] args: [[[-7811051307356131560, 1648420586873313002, -6846296687493726523, -8422746087891269182, 4826210766217018022, -7213350354792775054, -6884369668187685964, -3049336212063361297, -2533306905901105804, -8600407542072667435, 6989016086404929711, 2642913469597957988, -2213538430018240598, 155035492794154......, kwargs: {'timeout': 180} (api_request.py:62)

[2024-03-10T20:24:54.895Z] [2024-03-10 20:21:35 - DEBUG - ci_test]: (api_response) : (insert count: 2000, delete count: 0, upsert count: 0, timestamp: 448293003780685825, success count: 2000, err count: 0)  (api_request.py:37)

[2024-03-10T20:24:54.895Z] [2024-03-10 20:21:35 - DEBUG - ci_test]: (api_request)  : [Collection.load] args: [None, 1, 180], kwargs: {} (api_request.py:62)

[2024-03-10T20:24:54.895Z] [2024-03-10 20:24:36 - ERROR - pymilvus.decorators]: RPC error: [wait_for_loading_collection], <MilvusException: (code=1, message=wait for loading collection timeout, collection: QueryChecker__gxnVwxgP)>, <Time:{'RPC start': '2024-03-10 20:21:35.947045', 'RPC error': '2024-03-10 20:24:36.033606'}> (decorators.py:134)

[2024-03-10T20:24:54.895Z] [2024-03-10 20:24:36 - WARNING - pymilvus.decorators]: Retry timeout: 180s (decorators.py:100)

[2024-03-10T20:24:54.895Z] [2024-03-10 20:24:36 - ERROR - pymilvus.decorators]: RPC error: [load_collection], <MilvusException: (code=1, message=Retry timeout: 180s, message=wait for loading collection timeout, collection: QueryChecker__gxnVwxgP)>, <Time:{'RPC start': '2024-03-10 20:21:35.928577', 'RPC error': '2024-03-10 20:24:36.034013'}> (decorators.py:134)

[2024-03-10T20:24:54.895Z] [2024-03-10 20:24:36 - ERROR - ci_test]: Traceback (most recent call last):

[2024-03-10T20:24:54.895Z]   File "/home/jenkins/agent/workspace/tests/python_client/utils/api_request.py", line 32, in inner_wrapper

[2024-03-10T20:24:54.895Z]     res = func(*args, **_kwargs)

[2024-03-10T20:24:54.895Z]   File "/home/jenkins/agent/workspace/tests/python_client/utils/api_request.py", line 63, in api_request

[2024-03-10T20:24:54.895Z]     return func(*arg, **kwargs)

[2024-03-10T20:24:54.895Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/orm/collection.py", line 419, in load

[2024-03-10T20:24:54.895Z]     conn.load_collection(

[2024-03-10T20:24:54.895Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 135, in handler

[2024-03-10T20:24:54.895Z]     raise e from e

[2024-03-10T20:24:54.895Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 131, in handler

[2024-03-10T20:24:54.895Z]     return func(*args, **kwargs)

[2024-03-10T20:24:54.895Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 170, in handler

[2024-03-10T20:24:54.895Z]     return func(self, *args, **kwargs)

[2024-03-10T20:24:54.895Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 101, in handler

[2024-03-10T20:24:54.895Z]     raise MilvusException(

[2024-03-10T20:24:54.895Z] pymilvus.exceptions.MilvusException: <MilvusException: (code=1, message=Retry timeout: 180s, message=wait for loading collection timeout, collection: QueryChecker__gxnVwxgP)>

[2024-03-10T20:24:54.895Z]  (api_request.py:45)
```

### Expected Behavior

load success

### Steps To Reproduce

```markdown
the collection already be loaded
1. insert/query/search
2. release
3. load

failed at step 3
```

### Milvus Log

failed job: https://qa-jenkins.milvus.io/blue/organizations/jenkins/chaos-test-cron/detail/chaos-test-cron/12336/pipeline
log:
[artifacts-etcd-followers-pod-failure-12336-server-logs.tar.gz](https://github.com/milvus-io/milvus/files/14553023/artifacts-etcd-followers-pod-failure-12336-server-logs.tar.gz)

failed job: https://qa-jenkins.milvus.io/blue/organizations/jenkins/chaos-test-kafka-cron/detail/chaos-test-kafka-cron/12191/pipeline
log:
[artifacts-rootcoord-pod-failure-12191-server-logs.tar.gz](https://github.com/milvus-io/milvus/files/14553040/artifacts-rootcoord-pod-failure-12191-server-logs.tar.gz)

**Comment by zhuwenxing** [comment]
still reproduced in image tag 2.4-20240320-a07abba1-amd64
failed job: https://qa-jenkins.milvus.io/blue/organizations/jenkins/chaos-test-for-release-cron/detail/chaos-test-for-release-cron/11262/pipeline
log:
[artifacts-etcd-followers-pod-failure-11262-server-logs.tar.gz](https://github.com/milvus-io/milvus/files/14681508/artifacts-etcd-followers-pod-failure-11262-server-logs.tar.gz)

cluster: 4am, namespace: chaos-testing
pod info:
```
[2024-03-20T23:57:01.937Z] + kubectl get pods -o wide

[2024-03-20T23:57:01.939Z] + grep etcd-followers-pod-failure-11262

[2024-03-20T23:57:02.194Z] etcd-followers-pod-failure-11262-0                                1/1     Running       2 (15m ago)     33m     10.104.23.207   4am-node27   <none>           <none>

[2024-03-20T23:57:02.194Z] etcd-followers-pod-failure-11262-1                                1/1     Running       0               33m     10.104.18.193   4am-node25   <none>           <none>

[2024-03-20T23:57:02.194Z] etcd-followers-pod-failure-11262-2                                1/1     Running       0               33m     10.104.16.181   4am-node21   <none>           <none>

[2024-03-20T23:57:02.194Z] etcd-followers-pod-failure-11262-milvus-datacoord-b78bc4dfwdnqb   1/1     Running       0               33m     10.104.4.65     4am-node11   <none>           <none>

[2024-03-20T23:57:02.194Z] etcd-followers-pod-failure-11262-milvus-datanode-d48d5b969hjbmm   1/1     Running       0               33m     10.104.32.165   4am-node39   <none>           <none>

[2024-03-20T23:57:02.194Z] etcd-followers-pod-failure-11262-milvus-datanode-d48d5b969x5r5d   1/1     Running       0               33m     10.104.4.67     4am-node11   <none>           <none>

[2024-03-20T23:57:02.194Z] etcd-followers-pod-failure-11262-milvus-indexcoord-78c9fd4skpkt   1/1     Running       0               33m     10.104.23.203   4am-node27   <none>           <none>

[2024-03-20T23:57:02.194Z] etcd-followers-pod-failure-11262-milvus-indexnode-74874c7d9qw4p   1/1     Running       0               33m     10.104.32.166   4am-node39   <none>           <none>

[2024-03-20T23:57:02.194Z] etcd-followers-pod-failure-11262-milvus-indexnode-74874c7dql9t2   1/1     Running       0               33m     10.104.33.63    4am-node36   <none>           <none>

[2024-03-20T23:57:02.194Z] etcd-followers-pod-failure-11262-milvus-indexnode-74874c7dtbd5x   1/1     Running       0               33m     10.104.4.66     4am-node11   <none>           <none>

[2024-03-20T23:57:02.194Z] etcd-followers-pod-failure-11262-milvus-proxy-5cbb68649-9422p     1/1     Running       0               33m     10.104.32.168   4am-node39   <none>           <none>

[2024-03-20T23:57:02.194Z] etcd-followers-pod-failure-11262-milvus-querycoord-547874f7xfct   1/1     Running       0               33m     10.104.32.163   4am-node39   <none>           <none>

[2024-03-20T23:57:02.194Z] etcd-followers-pod-failure-11262-milvus-querynode-77f9c9f42n9dz   1/1     Running       0               33m     10.104.33.62    4am-node36   <none>           <none>

[2024-03-20T23:57:02.194Z] etcd-followers-pod-failure-11262-milvus-querynode-77f9c9f49gm9b   1/1     Running       0               33m     10.104.32.167   4am-node39   <none>           <none>

[2024-03-20T23:57:02.194Z] etcd-followers-pod-failure-11262-milvus-querynode-77f9c9f4hrgfr   1/1     Running       0               33m     10.104.23.202   4am-node27   <none>           <none>

[2024-03-20T23:57:02.194Z] etcd-followers-pod-failure-11262-milvus-rootcoord-58d5fd9dt6djm   1/1     Running       0               33m     10.104.32.164   4am-node39   <none>           <none>

[2024-03-20T23:57:02.194Z] etcd-followers-pod-failure-11262-minio-0                          1/1     Running       0               33m     10.104.16.177   4am-node21   <none>           <none>

[2024-03-20T23:57:02.194Z] etcd-followers-pod-failure-11262-minio-1                          1/1     Running       0               33m     10.104.23.206   4am-node27   <none>           <none>

[2024-03-20T23:57:02.194Z] etcd-followers-pod-failure-11262-minio-2                          1/1     Running       0               33m     10.104.32.173   4am-node39   <none>           <none>

[2024-03-20T23:57:02.194Z] etcd-followers-pod-failure-11262-minio-3                          1/1     Running       0               33m     10.104.20.82    4am-node22   <none>           <none>

[2024-03-20T23:57:02.194Z] etcd-followers-pod-failure-11262-pulsar-bookie-0                  1/1     Running       0               33m     10.104.29.156   4am-node35   <none>           <none>

[2024-03-20T23:57:02.194Z] etcd-followers-pod-failure-11262-pulsar-bookie-1                  1/1     Running       0               33m     10.104.19.152   4am-node28   <none>           <none>

[2024-03-20T23:57:02.194Z] etcd-followers-pod-failure-11262-pulsar-bookie-init-45r82         0/1     Completed     0               33m     10.104.6.87     4am-node13   <none>           <none>

[2024-03-20T23:57:02.194Z] etcd-followers-pod-failure-11262-pulsar-broker-0                  1/1     Running       0               33m     10.104.6.90     4am-node13   <none>           <none>

[2024-03-20T23:57:02.194Z] etcd-followers-pod-failure-11262-pulsar-proxy-0                   1/1     Running       0               33m     10.104.1.51     4am-node10   <none>           <none>

[2024-03-20T23:57:02.194Z] etcd-followers-pod-failure-11262-pulsar-pulsar-init-wbpmt         0/1     Completed     0               33m     10.104.6.86     4am-node13   <none>           <none>

[2024-03-20T23:57:02.194Z] etcd-followers-pod-failure-11262-pulsar-recovery-0                1/1     Running       0               33m     10.104.32.170   4am-node39   <none>           <none>

[2024-03-20T23:57:02.194Z] etcd-followers-pod-failure-11262-pulsar-zookeeper-0               1/1     Running       0               33m     10.104.32.172   4am-node39   <none>           <none>
```

**Comment by zhuwenxing** [comment]
standalone pod kill
failed job: https://qa-jenkins.milvus.io/blue/organizations/jenkins/chaos-test-kafka-for-release-cron/detail/chaos-test-kafka-for-release-cron/11727/pipeline/

log:
[artifacts-standalone-pod-kill-11727-server-logs.tar.gz](https://github.com/milvus-io/milvus/files/14693495/artifacts-standalone-pod-kill-11727-server-logs.tar.gz)

**Comment by zhuwenxing** [comment]
many collections loaded failed in datanode pod kill chaos test
failed job: https://qa-jenkins.milvus.io/blue/organizations/jenkins/chaos-test-kafka-for-release-cron/detail/chaos-test-kafka-for-release-cron/11729/pipeline
log:
[artifacts-datanode-pod-kill-11729-server-logs.tar.gz](https://github.com/milvus-io/milvus/files/14693536/artifacts-datanode-pod-kill-11729-server-logs.tar.gz)

**Comment by weiliu1031** [comment]
> many collections loaded failed in datanode pod kill chaos test failed job: https://qa-jenkins.milvus.io/blue/organizations/jenkins/chaos-test-kafka-for-release-cron/detail/chaos-test-kafka-for-release-cron/11729/pipeline log: [artifacts-datanode-pod-kill-11729-server-logs.tar.gz](https://github.com/milvus-io/milvus/files/14693536/artifacts-datanode-pod-kill-11729-server-logs.tar.gz)

load segment may stuck at build tmp index, @cqy123456 is trying to work on this, make it execute in async.
![image](https://github.com/milvus-io/milvus/assets/108661493/5a3e0a09-0a53-45e5-9d2e-b26858e62155)

to verify the case, try to disable tmp index by set `queryNode.segcore.interimIndex.enableIndex` to `false`

**Comment by xiaofan-luan** [comment]
that seems to take too long, I would expect less than 10s

**Comment by cqy123456** [comment]
Growing index and Binlog index share same knowhere build index pool. The pool size is 0.5 * cpu number. When push too much tasks into the pool, load segment cost will include the time of queueing in the pool.
The temporary solution is that binlog will further decide whether to build a binlig index by  the number of waiting tasks nukber in the knowhere build pool.  The final solution is to use a async function to build binlog index.

**Comment by zhuwenxing** [comment]
failed job: https://qa-jenkins.milvus.io/blue/organizations/jenkins/chaos-test-kafka-for-release-cron/detail/chaos-test-kafka-for-release-cron/12373/pipeline
log:
[artifacts-datanode-pod-kill-12373-server-logs.tar.gz](https://github.com/milvus-io/milvus/files/14954916/artifacts-datanode-pod-kill-12373-server-logs.tar.gz)

almost all collections load failed with timeout
```

[2024-04-12T03:50:59.554Z] [2024-04-12 03:50:59 - INFO - ci_test]: [test][2024-04-12T03:50:56Z] [3.07894046s] Hello_Milvus flush -> None (wrapper.py:30)

[2024-04-12T03:50:59.555Z] [2024-04-12 03:50:59 - INFO - ci_test]: assert entities: 8000 (test_data_persistence.py:93)

[2024-04-12T03:54:06.064Z] [2024-04-12 03:53:59 - ERROR - pymilvus.decorators]: RPC error: [wait_for_loading_collection], <MilvusException: (code=1, message=wait for loading collection timeout, collection: Hello_Milvus)>, <Time:{'RPC start': '2024-04-12 03:50:59.207771', 'RPC error': '2024-04-12 03:53:59.321769'}> (decorators.py:146)

[2024-04-12T03:54:06.064Z] [2024-04-12 03:53:59 - WARNING - pymilvus.decorators]: Retry timeout: 180s (decorators.py:106)

[2024-04-12T03:54:06.064Z] [2024-04-12 03:53:59 - ERROR - pymilvus.decorators]: RPC error: [load_collection], <MilvusException: (code=1, message=Retry timeout: 180s, message=wait for loading collection timeout, collection: Hello_Milvus)>, <Time:{'RPC start': '2024-04-12 03:50:59.145642', 'RPC error': '2024-04-12 03:53:59.322391'}> (decorators.py:146)

[2024-04-12T03:54:06.064Z] [2024-04-12 03:53:59 - ERROR - ci_test]: Traceback (most recent call last):

[2024-04-12T03:54:06.064Z]   File "/home/jenkins/agent/workspace/tests/python_client/utils/api_request.py", line 32, in inner_wrapper

[2024-04-12T03:54:06.064Z]     res = func(*args, **_kwargs)

[2024-04-12T03:54:06.064Z]   File "/home/jenkins/agent/workspace/tests/python_client/utils/api_request.py", line 63, in api_request

[2024-04-12T03:54:06.064Z]     return func(*arg, **kwargs)

[2024-04-12T03:54:06.064Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/orm/collection.py", line 422, in load

[2024-04-12T03:54:06.064Z]     conn.load_collection(

[2024-04-12T03:54:06.064Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 147, in handler

[2024-04-12T03:54:06.064Z]     raise e from e

[2024-04-12T03:54:06.064Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 143, in handler

[2024-04-12T03:54:06.064Z]     return func(*args, **kwargs)

[2024-04-12T03:54:06.064Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 182, in handler

[2024-04-12T03:54:06.064Z]     return func(self, *args, **kwargs)

[2024-04-12T03:54:06.064Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 107, in handler

[2024-04-12T03:54:06.064Z]     raise MilvusException(

[2024-04-12T03:54:06.064Z] pymilvus.exceptions.MilvusException: <MilvusException: (code=1, message=Retry timeout: 180s, message=wait for loading collection timeout, collection: Hello_Milvus)>

[2024-04-12T03:54:06.064Z]  (api_request.py:45)

[2024-04-12T03:54:06.064Z] [2024-04-12 03:53:59 - ERROR - ci_test]: (api_response) : <MilvusException: (code=1, message=Retry timeout: 180s, message=wait for loading collection timeout, collection: Hello_Milvus)> (api_request.py:46)
```

but from the grafana, the latency looks well
![image](https://github.com/milvus-io/milvus/assets/12268675/c39d441b-2f83-40ec-ba09-0870497f9278)

cc @weiliu1031

**Comment by yanliang567** [comment]
already fixed in milvus 2.4.1
